ydb/core/kqp/runtime/kqp_read_actor.cpp

```diff
@@ -1385,6 +1392,10 @@ class TKqpReadActor : public TActorBootstrapped<TKqpReadActor>, public NYql::NDq
             }
         }
         TBase::PassAway();
+
+        if (ReadActorSpan) {
+            ReadActorSpan.End();
+        }
```

Comment:
```suggestion
        ReadActorSpan.End();
```
seems like this should be enough as `End()` itself does `if (*this)` check